### PR TITLE
Codebridge version history: force a new version on start over

### PR DESCRIPTION
**Pull request by molly-moen**
Previously, when we restored any version except the start version, it would create a new version in the version history due to the way restore works. This PR does the same thing when restoring the "initial version", so that users don't potentially lose the most recent edits before they restored the initial version.

We already had the logic on the sources store to force create a new version (see #60162), but didn't have it plumbed through the Project Manager. This does that (defaulting to false), and uses the new boolean. I also previously missed flushing save before previewing the initial version (so we save any code the user added before viewing the old version), this PR adds that too.

## Links

- jira ticket: [CT-778](https://codedotorg.atlassian.net/browse/CT-778)

## Testing story
Tested locally and added appropriate Project Manager unit tests.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx [hunk 1]
```diff
@@ -118,8 +118,15 @@ const VersionHistoryDropdown: React.FunctionComponent<
   );
 
   const startOver = useCallback(() => {
-    // TODO: confirm
-    dispatch(setAndSaveProjectSource(startSource));
+    // We force a new version on start over so the user doesn't lose their recent edits.
```
Comment (ebeastlake) [inline apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx:124]: You're saying that when the user starts over, we don't wipe the version history? For some reason, I assumed restoring the initial version wouldn't wipe the version history, but starting over would, but I guess those are the same thing 😅

File: apps/src/lab2/projects/ProjectManager.ts [hunk 2]
```diff
@@ -168,17 +168,22 @@ export default class ProjectManager {
    * If a save is already enqueued, update this.sourceToSave with the given source.
    * @param sources ProjectSources: the source to save.
    * @param forceSave boolean: if the save should happen immediately
```
Comment (ebeastlake) [inline apps/src/lab2/projects/ProjectManager.ts:170]: "Immediately" = save before you create a new version?

**Comment by ebeastlake** [comment]
A couple conceptual questions I'm curious about, but code looks good to me!

**Review by ebeastlake — APPROVED** [review]

File: apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx (hunk 1 above)
Comment (molly-moen) [reply apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx:124]: Yeah it's a bit confusing, "start over" restores the initial sources, but does not wipe your version history.

File: apps/src/lab2/projects/ProjectManager.ts (hunk 2 above)
Comment (molly-moen) [reply apps/src/lab2/projects/ProjectManager.ts:170]: This method is what we call when there's a change on the project. The method will trigger a save within the next 30 seconds unless `forceSave` is true, in which case we trigger a save immediately. The new parameter `forceNewVersion` will force a new version to be created (by default we create a new version every 15 minutes). If both are true, we immediately save a new version.

**Pull request merged**